.github/DEVELOPER.md

```diff
@@ -0,0 +1,139 @@
+# CI/CD Workflow Guide
+
+TODO: Add a description of the CI/CD workflow and its components.
+
+### Overview
+
+Our CI/CD pipeline tests and builds our project across multiple languages, versions, and environments. This guide outlines the key components and processes of our workflow.
+
+### Workflow Triggers
+
+*   Push to `main` branch
+*   Pull requests
+*   Scheduled runs (daily)
+*   Manual trigger (workflow_dispatch)
+*   Pull request with `Core changes` label
+
+### Language-Specific Workflows
+
+* Python (python.yml)
+
+* Node.js (node.yml)
+
+* Java (java.yml)
+
+Each language has its own workflow file with similar structure but language-specific steps.
+
+### Shared Components
+
+#### Run Field
+
+The `run` field in both engine-matrix.json and build-matrix.json is used to specify when a particular configuration should be executed:
```

Comment:
I would start stating that we store the matrices that we load to the workflows under the json_matrices folder